File: programmers/level_2/짝지어제거하기.py
def solution(s):
    stack = []
    # 문자열의 갯수가 홀수면 볼필요없음
    if len(s) % 2:
        return 0

    for i in range(len(s)):
        if stack and stack[-1] == s[i]:
            stack.pop()
            continue

        stack.append(s[i])

        # 남은 문자열의 갯수가 stack의 길이보다 크면 볼필요없음
        if len(s) - i < len(stack):
            return 0

    return 1

File: programmers/level_2/짝지어제거하기.js
function solution(s) {
  // 문자열의 갯수가 홀수면 0
  if (s.length % 2 != 0) return 0;

  var sArr = [...s];
  var stack = [];

  for (var i = 0; i < sArr.length; i++) {
    if (stack && stack[stack.length - 1] === sArr[i]) {
      stack.pop();
      continue;
    }
    stack.push(sArr[i]);

    // 남은 문자열의 갯수가 stack의 길이보다 작으면 0
    if (sArr.length - i < stack.length) return 0;
  }

  return 1;
}
